File: backend/BookModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace DefaultNamespace
{
    public class BookModel
    {
        [Key] 
        public int Id { get; set; } 

        [Required] 
        public string Title { get; set; } 

        [Required] 
        public string Author { get; set; } 

        [Required] 
        public string Publisher { get; set; } 

        [Required] 
        public DateTime PublicationDate { get; set; } 

        [Required] 
        public string ISBN { get; set; } 

        public string Edition { get; set; } 

        [Required] 
        public string Language { get; set; } 


     /*   public BookModel(int id, string title, string author, string publisher, DateTime publicationDate, string isbn, string edition, string language)
        {
            this.Id = id;
            this.Title = title;
            this.Author = author;
            this.Publisher = publisher;
            this.PublicationDate = publicationDate;
            this.ISBN = isbn;
            this.Edition = edition;
            this.Language = language;
        } */
    }
}

File: frontend-vue/src/models/bookModel.ts
export interface BookModel{

  /** The unique identifier for the book. */
  id: number;

  /** The title of the book. */
  title: string;

  /** The author of the book. */
  author: string;

  /** The publisher of the book. */
  publisher: string;

  /** The date when the book was published. */
  publicationDate: Date;

  /** The ISBN number of the book. */
  isbn: string;

  /** The edition of the book. */
  edition?: string; // Optional since the original property is not required

  /** The language the book is written in. */
  language: string;

}
